### PR TITLE
Removed `uninitialized_statics` field from `Memory` struct in miri

### PR DESCRIPTION
**Pull request by alexreg**
based on #51110 

r? @oli-obk

CC @eddyb

### CODE REVIEW

**Comment by alexreg** [comment]
@oli-obk All tests are passing locally, just waiting on Travis here. See my message on Discord re `MarkStatic`, but I think this is ready to review now and merge as soon as the other PR goes through.

**Comment by oli-obk** [comment]
Could you also check how much of an impact this has on the miri submodule?

File: src/librustc_mir/interpret/memory.rs [hunk 1]
```diff
@@ -396,26 +376,23 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
                         MemoryKind::Machine(m) => format!(" ({:?})", m),
                     }),
                     // uninitialized static alloc?
```
Comment (oli-obk) [inline src/librustc_mir/interpret/memory.rs]: leftover comment

File: src/librustc_mir/interpret/memory.rs [hunk 2]
```diff
@@ -83,19 +77,16 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
...
+        kind: MemoryKind<M::MemoryKinds>,
     ) -> EvalResult<'tcx, AllocId> {
         let id = self.tcx.alloc_map.lock().reserve();
         M::add_lock(self, id);
         match kind {
-            Some(kind @ MemoryKind::Stack) |
-            Some(kind @ MemoryKind::Machine(_)) => {
+            kind @ MemoryKind::Stack |
+            kind @ MemoryKind::Machine(_) => {
                 self.alloc_map.insert(id, alloc);
```
Comment (oli-obk) [inline src/librustc_mir/interpret/memory.rs]: I think these two maps can now be merged into one

**Review by oli-obk — COMMENTED** [review]
That was less fallout than I feared!

File: src/librustc_mir/interpret/memory.rs (hunk 2 above)
Comment (alexreg) [reply src/librustc_mir/interpret/memory.rs]: Oops, yep.

**Comment by alexreg** [comment]
> Could you also check how much of an impact this has on the miri submodule?

How do you mean, exactly?

**Comment by alexreg** [comment]
What's the Travis test failure due to, do you think? Not happening on my machine.

**Comment by oli-obk** [comment]
Well... I guess we first need to fix miri on nightly before it would make sense to adjust it to the changes of this PR

**Comment by oli-obk** [comment]
Not sure. Can you provide a `TRACE=rustc_mir::interpret`?

**Comment by alexreg** [comment]
@oli-obk The test passes locally though so...

**Comment by alexreg** [comment]
Oh, the errors are to do with `tried to read foreign (extern) static` messages, which is from my other PR. Let's wait until that merges, and see if they still appear. I'm not quite sure on the origin, but if a fix is needed, it's trivial (it's literally just changing the expected messages).

**Comment by pietroalbini** [comment]
Blocked on https://github.com/rust-lang/rust/pull/51110.

**Comment by alexreg** [comment]
Err, looks like a transitory issue with CI.

**Comment by pietroalbini** [comment]
Manually restarted the build.

**Comment by alexreg** [comment]
Okay, that’s the same error I’m getting in the PR blocking this. Looks like something changed in master with the rebase.

**Comment by alexreg** [comment]
@oli-obk Okay, I've rebased over master now that #51110 has merged. I think we're ready to merge this too now!

File: src/librustc_mir/interpret/eval_context.rs [hunk 3]
```diff
@@ -9,14 +10,24 @@ use rustc::ty::subst::{Subst, Substs};
 use rustc::ty::{self, Ty, TyCtxt, TypeAndMut};
 use rustc::ty::query::TyCtxtAt;
 use rustc_data_structures::indexed_vec::{IndexVec, Idx};
+<<<<<<< HEAD
 use rustc::mir::interpret::FrameInfo;
 use syntax::codemap::{self, Span};
 use syntax::ast::Mutability;
+||||||| parent of a2543471ec... Removed `uninitialized_statics` field from `Memory` struct in miri.
```
Comment (oli-obk) [inline src/librustc_mir/interpret/eval_context.rs]: rebase fallout

File: src/librustc_mir/interpret/memory.rs [hunk 4]
```diff
@@ -6,12 +6,24 @@ use rustc::ty::Instance;
 use rustc::ty::ParamEnv;
 use rustc::ty::query::TyCtxtAt;
 use rustc::ty::layout::{self, Align, TargetDataLayout, Size};
+<<<<<<< HEAD
 use syntax::ast::Mutability;
 
 use rustc_data_structures::fx::{FxHashSet, FxHashMap};
+||||||| parent of a2543471ec... Removed `uninitialized_statics` field from `Memory` struct in miri.
```
Comment (oli-obk) [inline src/librustc_mir/interpret/memory.rs]: more rebase fallout

Comment (alexreg) [reply src/librustc_mir/interpret/memory.rs]: I fixed this though! WTF?

**Pull request merged**